flow/JsonString.h

```diff
@@ -0,0 +1,142 @@
+		JsonString& append( const std::string& name, long long int value );
+		JsonString& append( const std::string& name, long long unsigned int value );
+		JsonString& append( const std::string& name, int value );
+		JsonString& append( const std::string& name, unsigned value );
+		JsonString& append( const std::string& name, bool value );
+		JsonString& append( const std::string& name, const JsonString& value );
+		JsonString& append( const std::string& name, const JsonStringArray& value );
+
+		JsonString& append( const std::string& value );
+		JsonString& append( const char* value );
+		JsonString& append( const JsonStringArray& value );
+
+		JsonString& append( double value );
+		JsonString& append( long int value );
+		JsonString& append( long unsigned int value );
+		JsonString& append( long long int value );
+		JsonString& append( long long unsigned int value );
+		JsonString& append( int value );
+		JsonString& append( unsigned value );
+		JsonString& append( bool value );
+		JsonString& append( const JsonString& value );
+
+
+		JsonStringSetter& operator[]( const std::string& name );
+		JsonStringSetter& operator[]( const char* name );
+
+		int compare( const JsonString& jsonString ) const;
+		bool equals( const JsonString& jsonString ) const;
+		bool operator==( const JsonString& jsonString ) const { return equals(jsonString); }
+		bool operator!=( const JsonString& jsonString ) const { return !equals(jsonString); }
+
+		JsonString& copy( const JsonString& jsonString );
+		JsonString& operator=( const JsonString& jsonString );
+
+		std::string	getJson() const;
+		void	setJson(const std::string& jsonText);
+
+		size_t	getLength() const;
+		size_t	getNameTotal() const;
+
+		JsonString& clear();
+		bool	empty() const;
+
+		bool	isPresent(const std::string& name) const;
+
+		static uint32_t hash32( const std::string& name );
+		static JsonString makeMessage(const char *name, const char *description);
+
+	protected:
+		void hashName( const std::string& name);
+		JsonString& appendImpl( const std::string& name, const std::string& value, bool quote);
+		JsonString& appendImpl( const std::string& value, bool quote);
+
+	protected:
+		std::string		_jsonText;
+		std::unordered_set<std::string>	_keyNames;
+
+		// Uneditted text
+		const std::string&	getJsonText() const;
+};
+
+// Make protected because no virtual destructor
+class JsonStringArray : protected std::vector<JsonString>
+{
+	typedef JsonString T;
+	typedef std::vector<JsonString> vector;
+	public:
+		using vector::push_back;
+		using vector::operator[];
+		using vector::begin;
+		using vector::end;
+		using vector::size;
+		using vector::empty;
+		using vector::clear;
+	public:
+		JsonStringArray();
+		JsonStringArray( const JsonStringArray& jsonStringArray);
+		virtual ~JsonStringArray();
+};
+
+class JsonStringSetter {
+	public:
+
+	JsonStringSetter( JsonString& jsonString, const std::string& name );
+
+	JsonStringSetter& operator=( const std::string& value );
```

Comment:
These operator= overloads can probably all be handled by a single templated definition of operator=.